src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -302,139 +308,148 @@ def _execute(self, rseList):
             self.plineCounters[pline.name]['rsesCleaned'], \
             self.plineCounters[pline.name]['filesDeletedSuccess']
 
+    def resetCounters(self, rse):
+        """
+        Reset all counters for an RSE object and initialize structure if needed.
+        :param rse: The RSE object whose counters need to be reset
+        """
+        # Initialize the basic structure if it doesn't exist
+        rse.setdefault('dirs', {})
+        rse.setdefault('counters', {})
+
+        # Initialize directory structure
+        rse['dirs'].setdefault('toDelete', set())
+        rse['dirs'].setdefault('deletedSuccess', set())
+        rse['dirs'].setdefault('deletedFail', set())
+
+        # Initialize counter structure
+        rse['counters'].setdefault('dirsDeletedSuccess', 0)
+        rse['counters'].setdefault('dirsDeletedFail', 0)
+        rse['counters'].setdefault('filesDeletedSuccess', 0)
+        rse['counters'].setdefault('filesDeletedFail', 0)
+        rse['counters'].setdefault('gfalErrors', {})
+
+        self.logger.info("Reset all counters for RSE: %s", rse['name'])
+        return rse
+
     # @profile
     def cleanRSE(self, rse):
         """
-        The method to implement the actual deletion of files for an RSE.
-        :param rse: MSUnmergedRSE object to be cleaned
-        :return:    The MSUnmergedRSE object
+        Optimized cleanup method that minimizes directory listing operations
+        and handles storage technologies that support recursive directory deletion.
```

Comment:
I suggest to put back description of input parameter and return object. Since `rse` is dictionary object whose size and scalability is not-known why it was a decision to return copy of this object instead of making in-place changes?

Reply:
I shouldn't have deleted that part of the docstring and I am adding it back. For the second comment, I will write it at the main body of the PR.